### PR TITLE
fix: [sc-114420] The last line of sync & transposed contains NaN

### PR DESCRIPTION
**Pull request by alexbourret**

### CODE REVIEW

File: custom-recipes/pi-system-transpose/recipe.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def get_latest_data_at_timestamp(file_handles, timestamp):
             current_values_cache[cache_index] = next_values_cache[cache_index]
             line = file_handles[attribute_path].readline()
             if not line:
-                current_values_cache[cache_index] = "NaN"
+                current_values_cache[cache_index] = next_values_cache[cache_index]
```
Comment (liamlynch-data) [inline custom-recipes/pi-system-transpose/recipe.py:75]: I think we should just delete this line completely - the assignment `current_values_cache[cache_index] = next_values_cache[cache_index]`  is already made above, so this doesn't do anything.
So if I'm not going crazy things should still work without it.

More generally, I found that whole function a bit hard to follow (I didn't find the code self documenting and I'm not sure how I would write it to be so - so this is not a criticism, it is tricky for a reason!). All I can suggest is the following that would have helped me personally:

1. Renaming `cache_index` to `attrib_index` (removes any distracting doubt about what it means)

2. Renaming the `timestamp` argument to `seek_timestamp`

3. adding the following comment or something like it:

_For each attribute, keep track of the last timestamp and value read in next_timestamps_cache[attrib_index] and next_timestamps_cache[attrib_index]_


_We move along to the seek_timestamp - so if the next_timestamps_cache[attrib_index] does not yet exceed the seek_timestamp, we keep reading lines. As we do this we maintain current_timestamps_cache[attrib_index] and current_values_cache[attrib_index] as the values one behind the reading head - so when we stop reading, these are the correct values to return (at the end of the file they will be set from the last line read, which should also be correct)_

**Review by liamlynch-data — COMMENTED** [review]
The fix works - I have tested it - and the clean up and error handing changes are good. 
I just have a couple of suggestions about making it easier to understand.

File: custom-recipes/pi-system-transpose/recipe.py (hunk 1 above)
Comment (alexbourret) [reply custom-recipes/pi-system-transpose/recipe.py]: 👍 It is indeed clearer... I made the suggested changes in [2eeebdc](https://github.com/dataiku/dss-plugin-osisoft/pull/19/commits/2eeebdc9d2190ecd613f57c2e43c1cb7fa634116)

Comment (liamlynch-data) [reply custom-recipes/pi-system-transpose/recipe.py]: Thanks! I still think it needs a comment though - something short just above the loop like this should be enough:
_Continue along file till just passed the seek_timestamp - current_timestamps_cache & current_values_cache should then have the values we read for the attribute at the timestamp we want_

Comment (liamlynch-data) [reply custom-recipes/pi-system-transpose/recipe.py]: added here in tail PR so fine https://github.com/dataiku/dss-plugin-osisoft/pull/20/commits/7167b1aec2f0918531768da7806a9aa7683b867f

**Review by liamlynch-data — APPROVED** [review]
All good 👍
